packages/strapi-design-system/src/v2/Breadcrumbs/CrumbLink.js

```diff
@@ -23,7 +23,14 @@ export const CrumbLink = ({ children, ...props }) => <StyledLink {...props}>{chi
 
 CrumbLink.displayName = 'CrumbLink';
 
+const requiredPropsCheck = (props, propName, componentName) => {
+  if (!props.href && !props.to && !props.onClick) {
+    return new Error(`One of 'href', 'to' or 'onClick' is required by '${componentName}' component.`);
+  }
+};
+
 CrumbLink.propTypes = {
   children: PropTypes.string.isRequired,
-  href: PropTypes.string.isRequired,
+  href: requiredPropsCheck,
+  to: requiredPropsCheck,
```

Comment:
I'm wondering if we should get rid of this for the v2 version. I believe this comes from the `Link` refactoring, but we should either unify it, or use the v1, where we can not at the moment.

What do you think?

Reply:
Do you mean remove href/to/onClick to be required at all?
I added this fix because saying that href is required doesn't work:
- We can use as={NavLink} for this component, which needs a to prop and not an href which caused proptypes errors
- We need to use this component with as="button" sometimes (e.g. Media Library in CM, we are using a button to change folder and not a link), in this case there won't be href or to prop, only an onClick one
- It is really similar to FolderCard usage which can accept link or button (and link can be with href or to - React side)

maybe I'm missing something? 👀

Reply:
Phew, I see and didn't think of the `as="button"` case. Could we remove the `href=""` prop then, if we know it won't be needed (I guess that was my mistake initially - I thought we are using href)?

Reply:
yes sure, I just thought href could be useful one day but we could update it then if needed 👌

Reply:
Now that I think about it `Link` v2 doesn't require either, do shouldn't make them required at all for consistency or require them in `Link` v2 component too? 
<img width="381" alt="image" src="https://user-images.githubusercontent.com/71838159/181034250-69caf273-076e-44b5-b7be-3c1465e9b633.png">

Reply:
What happens to a Link component without `href`? Does it render as a span or does it become an invalid element?

Reply:
I would say to keep one of them required, in any case I find it helpful for developers, wdyt?
about href I would still keep it since all DS link components can be used as external link, I would find it weird to force NavLink or button, wdyt?

Reply:
Phew, I just realized I'm lost on where we want to go regarding `Link` and `NavLink`. @HichamELBSI I'd need your opinion here and try to summarize:

- `CrumbLink` will always be used with `as={NavLink}` (`NavLink` refers to `react-router-dom`)
- because of that, `CrumbLink` needs a `to` and a `href` prop: `to` is used by `NavLink`, `href` is used by `Link@v2`
- I'd like to get rid of one of these, because it feels like we have an inconsistency which is present in many places and I feel like we should settle for `to` everywhere in v2, since this is what `react-router-dom` is using and I can only imagine a very small number of cases where we'd not that one

But I'm not sure and I'm wondering what your thoughts are ...

Reply:
Just a thought after checking:
- `Link@v2` don't force developers to have an `href` prop because it can be a `NavLink`
- Btw BaseLink is offering a nice isExternal prop that is not showing in Links documentation 
- If we need to make any change on Links components I would say to do it in another PR since I'd like to move on quicker on Breacrumbs so I can also move on with the feature 🥲

- If you think @gu-stav `CrumbLink` shouldn't be use as an external link I can enforce only `to` or `onClick` prop (since it can be a button to as we can see in the ML modal in the CM)
- That won't prevent user to use BaseLink prop (`isExternal`) but they would have a prop error 🤷🏻‍♀️

Tbh I would like to move on with the PR, I'm not sure @HichamELBSI will have time to review it and I would like to finish the feature this sprint, wdyt?

Reply:
I'm not feeling comfortable to make a final decision on that now with another opinion, tbh, because I'm not sure I understand what the underlying idea is and where we want to go with `v2` ...

In the end, what I'd like to have is a clean interface for the `CrumbLink` component, which doesn't leave room for for ambiguity (which imho having `to` and `href` creates). Once the component is out and released, we won't be able to take the decision back and will have to live it for a long time and therefore I'd argue a solid decision is important here.

Reply:
The v2 components were created precisely in order to get rid of the `react-router-dom` dependency. Some users don't/can't use `react-router-dom`. For example, on `NextJS`, we need the href property. That's why our links must be render html links without modification on native properties. This is what we recommend in our documentation. 

![Screenshot 2022-07-27 at 11 07 42](https://user-images.githubusercontent.com/7756284/181209286-bc9950c2-9333-4ea4-ba62-c37f46c4151f.png)

https://github.com/strapi/design-system/issues/440

Does it make sense?

Reply:
As discussed in private messages, for now CrumbLink will have the same PropTypes definition as every link based on BaseLink:
- Enforcing only `to` or `href` would prevent Next developers to use the component without PropTypes errors
- In some cases CrumbLink needs to be a button (e.g. Modal navigation)
- We could redefine better PropTypes definition for links component, to be discussed during a DS meeting

@gu-stav @HichamELBSI